mcs/class/corlib/System.IO.Enumeration/FileSystemEnumerator.NotSupported.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections;
+
+namespace System.IO.Enumeration
+{
+	partial class FileSystemEnumerator<TResult>
```

Comment:
This does not match the API of the reference class

```
    public abstract class FileSystemEnumerator<TResult> : Runtime.ConstrainedExecution.CriticalFinalizerObject, Collections.Generic.IEnumerator<TResult>
    {
        public FileSystemEnumerator(string directory, EnumerationOptions options = null) { }

        protected virtual bool ShouldIncludeEntry(ref FileSystemEntry entry) { throw null; }
        protected virtual bool ShouldRecurseIntoEntry(ref FileSystemEntry entry) { throw null; }
        protected abstract TResult TransformEntry(ref FileSystemEntry entry);
        protected virtual void OnDirectoryFinished(ReadOnlySpan<char> directory) { throw null; }
        protected virtual bool ContinueOnError(int error) { throw null; }

        public TResult Current { get { throw null; } }
        object System.Collections.IEnumerator.Current { get { throw null; } }
        public bool MoveNext() { throw null; }
        public void Reset() { throw null; }
        public void Dispose() { throw null; }
        protected virtual void Dispose(bool disposing) { throw null; }
    }
```

Reply:
The API is provided by https://github.com/mono/corefx/blob/806496e9a4608b2a5a91c7fa0d3fe79cde616d44/src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.cs which is included in all corlib sources.

Reply:
I don't understand this is NSE version of the API but with slightly different shape